actor/builtin/paymentbroker/paymentbroker.go

```diff
@@ -255,6 +264,16 @@ func (pb *Actor) Redeem(vmctx exec.VMContext, payer address.Address, chid *types
 		// channel has been cancelled.
 		channel.Eol = channel.AgreedEol
 
+		// Mark the payment channel as redeemed and set the conditions
+		channel.Redeemed = true
+		if condition == nil {
+			channel.Conditions = nil
+		} else {
+			newConditions := *condition
+			newConditions.Params = append(newConditions.Params, redeemerConditionParams...)
```

Comment:
This should use the same params that were executed in the condition. All of this (except setting redeemed) should be set in `checkCondition`.